ci-automation/vms.sh

```diff
@@ -69,10 +69,24 @@ function vm_build() {
 
     [[ ${has_packet} -eq 1 ]] && [[ ${has_pxe} -eq 0 ]] && set -- 'pxe' "${@}"
 
-    for format; do
-        # keep compatibility with SDK scripts where "equinix_metal"
-        # remains unknown.
-        [ "${format}" = "equinix_metal" ] && format="packet"
+    # Convert platform names (also used to find the test scripts) to image formats they entail
+    formats="$*"
+    if echo "$formats" | tr ' ' '\n' | grep -q azure; then
+      formats=$(echo "$formats" | tr ' ' '\n' | sed '/azure.*/d')
+      formats+=" azure azure_gen2"
+    fi
+    if echo "$formats" | tr ' ' '\n' | grep -q '^vmware'; then
+      formats=$(echo "$formats" | tr ' ' '\n' | sed '/vmware.*/d')
+      formats+=" vmware vmware_insecure vmware_ova vmware_raw"
+    fi
+    if echo "$formats" | tr ' ' '\n' | grep -q -P '^(ami|aws)'; then
+      formats=$(echo "$formats" | tr ' ' '\n' | sed '/ami.*/d' | sed '/aws/d')
+      formats+=" ami ami_vmdk"
+    fi
+    # Keep compatibility with SDK scripts where "equinix_metal" remains unknown.
+    formats=$(echo "$formats" | tr ' ' '\n' | sed 's/equinix_metal /packet/g')
```

Comment:
```suggestion
    formats=$(echo "$formats" | tr ' ' '\n' | sed 's/equinix_metal/packet/g')
```

I don't think we'll match the `equinix_metal` otherwise.